### ISSUE TITLE
Fetching multiple access tokens silently only succeeds for one token

### ISSUE THREAD

**Issue by raikoss** [AzureAD/microsoft-authentication-library-for-js#722]
## I'm submitting a...

<pre><code>
[x] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Browser:
- [x] Chrome version XX
- [ ] Firefox version XX
- [ ] IE version XX
- [ ] Edge version XX
- [ ] Safari version XX

## Library version

<pre><code>
Library version: 1.0.0

</code></pre>

## Current behavior

When trying to fetch two access tokens silently at the same time after a login, only one of the responses actually includes the access token. When refreshing, one token gets fetched from the cache which makes the other token response succeed. 

## Expected behavior

Both token responses should succeed in getting the token.

## Minimal reproduction of the problem with instructions

```js
const msal = new Msal.UserAgentApplication(config);

// after a user has logged in
msal.acquireTokenSilent({ scopes: ['user.read'] })
.then(response => {
  // do something with access token
})

msal.acquireTokenSilent({ scopes: ['api://myapiguid/Read'] })
.then(response => {
  // do something with access token
})
```

When doing the above, either of the responses will not contain an access token, as they are being run at the same time. This used to work in the previous version of msal, 0.2.4 if I recall correctly. 

When doing the following workaround, the tokens are being acquired correctly: 

```js
msal.acquireTokenSilent({ scopes: ['user.read'] })
.then(response => {
  // do something with access token

  return msal.acquireTokenSilent({ scopes: ['api://myapiguid/Read'] });
})
.then(response => {
  // do something with access token
})
```

Which suggests that the tokens are not being successfully acquired when attempting to do so at the same time. 

Edit:
After some more testing, it correctly delays one request until the other one is done. I'm not really sure why, but before the issue happened as explained above. I'm not sure if something changed that affected this, so I'm leaving the issue open in case it happens again.

**Comment by ThomasJacob** [comment]
Another case, for multiple scope token graph scope and custom api scope
In case of first login it properly returns the token properly. But if there was a cache token available which has expired, After adding some delay after login the token gets returned but the token will be for the graph scope and not for the request one. If there is no delay it returns empty.

Can we move this to a high priority issue?

**Comment by DarylThayil** [comment]
@raikoss  @ThomasJacob  
Thank you for the feedback, this is a good call out of something we should fix. 
I am making a work item for the team, to make sure we can fetch multiple tokens concurrently.

Are either of you currently blocked? If I am thinking of the problem we need to solve in the code, it may not be a quick fix. 

@sameerag  / @pkanher617  I know we have some code in place that currently prevents that , so we may need to get creative to solve for this.

**Comment by DarylThayil** [comment]
@ThomasJacob  having a little trouble understanding your issue, is it related to the concurrent, or is it a separate issue that has to do with how we are caching ?

**Comment by raikoss** [comment]
@DarylThayil When I was testing this yesterday, there was a period where only one token got fetched, but as the day passed I somehow succeeded in getting both tokens. It seems pretty random, but in periods, as every test case I tried earlier that day failed, and every test case I tried later during the day succeeded. This also hasn't happened to me since I created this issue, strangely enough. 

Maybe there was something wrong with the client in Azure AD that somehow resolved itself? I'm sorry I don't have more to go on this, as it seems like it's a somewhat big issue if it persists.

**Comment by DarylThayil** [comment]
Closing without repro + solution for concurrent request being done serially.
We do have work slated to handle concurrent token requests in the future

**Comment by pkanher617** [comment]
@raikoss @ThomasJacob Re-opening because I was able to repro. I have added a fix for this in PR #768. Can you please check if this fixes your issue? I will push a patch as soon as I have confirmed that it sufficiently fixes the issue.

**Comment by raikoss** [comment]
The changes in the pull request seems to fix the issue.
